pkg/og/flameql/flameql.go

```diff
@@ -97,7 +97,7 @@ func ValidateAppName(n string) error {
 }
 
 func IsTagKeyRuneAllowed(r rune) bool {
-	return (r >= 'a' && r <= 'z') || (r >= 'A' && r <= 'Z') || (r >= '0' && r <= '9') || r == '_' || r == '.'
+	return (r >= 'a' && r <= 'z') || (r >= 'A' && r <= 'Z') || (r >= '0' && r <= '9') || r == '_' || r == '.' || r == '/'
```

Comment:
Is `IsTagKeyRuneAllowed` used for both label names and values? If yes, we probably want to change this. We should be more restrictive with label names and less with label values. E.g., we want to allow `-` in label values (a very common case if for services to be named `namespace-001/application-name`). If I am not mistaken, we currently allow any UTF-8 character (for label values) in the /Push path.

Reply:
After checking the code path more carefully, I see that label values aren't actually validated at all. If I'm not wrong, the `IsTagKeyRuneAllowed` is only used for:

- Label keys (via ValidateTagKey)
- App names (via ValidateAppName which uses `IsTagKeyRuneAllowed` plus some extra characters)

For our specific [issue](https://github.com/grafana/alloy/issues/2158) with app names being rejected when they contain slashes, we just need to allow slashes in `IsAppNameRuneAllowed`:
```golang
func IsAppNameRuneAllowed(r rune) bool {
    return r == '-' || r == '.' || r == '/' || IsTagKeyRuneAllowed(r)
}
```
Your point about label keys validation right but I'd say it's a separate enhancement we could tackle later (?) Related to prometheus allowing UTF-8 characters. Does it make sense?

Reply:
> Your point about label keys validation right but I'd say it's a separate enhancement we could tackle later (?) Related to prometheus allowing UTF-8 characters. Does it make sense?

To clarify, I didn't mean we should change label key validation right now, I was just saying it should be more restrictive than the one for label values in general. 

The change in 83b1701 makes sense to me!